test/jasmine/tests/colorbar_test.js

```diff
@@ -524,5 +524,44 @@ describe('Test colorbar:', function() {
             .catch(failTest)
             .then(done);
         });
+
+        it('creates the same colorbars attributes in newPlot and react', function(done) {
+            var z = [[1, 10], [100, 1000]];
+
+            var expectedAttrs = [];
+            var actualAttrs = [];
+
```

Comment:
The test looks great.
Non-blocking, but maybe you could add a getter function to avoid duplication:
Something like:
```js
function pushItemsTo(a) {
    var colorbars = d3.select(gd).selectAll('.colorbar');
    colorbars.selectAll('.cbfill').each(function() {
        var attrsForElem = {};
        for(var i = 0; i < this.attributes.length; i++) {
            var attr = this.attributes.item(i);
            attrsForElem[attr.name] = attr.value;
        }
        a.push(attrsForElem);
    });
}

Plotly.newPlot(gd, [{type: 'contour', z: z}])
.then(function() {
    pushItemsTo(expectedAttrs);

    return Plotly.newPlot(gd, [{type: 'heatmap', z: z}])
    .then(function() {
        return Plotly.react(gd, [{type: 'contour', z: z}]);
    });
})
.then(function() {
    pushItemsTo(actualAttrs);

    expect(actualAttrs).toEqual(expectedAttrs);
})
```

Reply:
Done :slightly_smiling_face: